src/kOS/Suffixed/StageValues.cs

```diff
@@ -31,81 +35,80 @@ private void InitializeSuffixes()
 
         private Lexicon GetResourceDictionary()
         {
-            var resources = shared.Vessel.GetActiveResources();
             var toReturn = new Lexicon();
-
-            foreach (var resource in resources)
+            var defs = PartResourceLibrary.Instance.resourceDefinitions;
+            foreach (var def in defs)
             {
-                toReturn.Add(new StringValue(resource.info.name), new ActiveResourceValue(resource, shared));
+                toReturn.Add(new StringValue(def.name), new ActiveResourceValue(shared.Vessel, def.id, shared));
             }
 
             return toReturn;
         }
 
         public override ISuffixResult GetSuffix(string suffixName)
         {
-            if (!IsResource(suffixName))
+            string fixedName;
+            if (!Utils.IsResource(suffixName, out fixedName))
             {
                 return base.GetSuffix(suffixName);
             }
 
-            var resourceAmount = GetResourceOfCurrentStage(suffixName);
+            var resourceAmount = GetResourceOfCurrentStage(fixedName);
             return new SuffixResult(ScalarValue.Create(resourceAmount.HasValue ? resourceAmount.Value : 0.0));
         }
 
-        private bool IsResource(string suffixName)
-        {
-            return PartResourceLibrary.Instance.resourceDefinitions.Any(
-                pr => string.Equals(pr.name, suffixName, StringComparison.CurrentCultureIgnoreCase));
-        }
-
         private double? GetResourceOfCurrentStage(string resourceName)
         {
-            PartResourceDefinition resourceDef = PartResourceLibrary.Instance.resourceDefinitions.FirstOrDefault(pr => string.Equals(pr.name, resourceName, StringComparison.CurrentCultureIgnoreCase));
+            PartResourceDefinition resourceDef = PartResourceLibrary.Instance.resourceDefinitions[resourceName];
+
+            double total = 0;
+            double capacity = 0;
 
             if (resourceDef == null)
             {
                 throw new KOSInvalidArgumentException("STAGE", resourceName, "The resource definition could not be found");
             }
 
-            var list = new List<PartResource>();
-            if (resourceDef.resourceFlowMode == ResourceFlowMode.STACK_PRIORITY_SEARCH)
+            CreatePartSet();
+
+            partSet.GetConnectedResourceTotals(resourceDef.id, out total, out capacity, true);
+
+            return Math.Round(total, 2);
+        }
+
+        public void CreatePartSet()
+        {
+            // The following replicates the logic in KSP.UI.Screens.ResourceDisplay.CreateResourceList
+            // We're creating the set every time because it doesn't pay attention to the various events
+            // that would tell us that the old partset is no longer valid.
+            partHash.Clear();
+            int vstgComp = shared.Vessel.currentStage - 2;
+            var parts = shared.Vessel.Parts;
+            global::Part part;
+            for (int i = parts.Count; i-- > 0;)
```

Comment:
I think this would be a lot clearer what your intent was: 
``for (int i = parts.Count - 1 ; i >= 0 ; --i)``
That makes it a lot more straightforward the this is a loop that goes from n-1 through 0.  Right now that behavior is a bit obfuscated behind the side-effecting of the boolean check.  The initializer says ``parts.Count`` despite the fact that the first iteration will actually be using ``parts.Count-1`` due to that side-effect.  The check says ``>0`` despite the fact that zero will be included within the iterations (because it checks the value of ``i`` prior to the decrement, ``>0`` *really* ends up meaning more like ``>=0``.)